joerd/check.py

```diff
@@ -19,6 +20,21 @@ def is_zip(tmp):
     return False
 
 
+def is_tar_gz(tmp):
+    """
+    Returns True if the NamedTemporaryFile given as the argument appears to be
+    a GZip-encoded TAR file.
+    """
+
+    try:
+        tar = tarfile.open(tmp.name, mode='r:gz', errorlevel=2)
+        names = [info.name for info in tar]
```

Comment:
I see. I'm not familiar with the details of tarballs, but I agree that this would probably only exercise a part of it, and that the names would likely be indexed somehow. Would this sort of thing ideally be checked by a checksum?

If we wanted to be more memory efficient, we'd just iterate through without collecting the names in a list. But, I don't think it's really worth worrying about.